server/chat-plugins/laddertours.ts

```diff
@@ -224,34 +224,41 @@ export class LadderTracker {
 	async getLeaderboard(display?: boolean) {
 		const url = `https://pokemonshowdown.com/ladder/${this.format}.json?prefix=${this.prefix}`;
 		const leaderboard: LeaderboardEntry[] = [];
-		try {
-			const response = await Net(url).get().then(JSON.parse);
-			this.leaderboard.lookup = new Map();
-			for (const data of response.toplist) {
-				// TODO: move the rounding until later
-				const entry: LeaderboardEntry = {
-					name: data.username,
-					elo: Math.round(data.elo),
-					gxe: data.gxe,
-					glicko: Math.round(data.rpr),
-					glickodev: Math.round(data.rprd),
-				};
-				this.leaderboard.lookup.set(data.userid, entry);
-				if (!data.userid.startsWith(this.prefix)) continue;
-				entry.rank = leaderboard.length + 1;
-				leaderboard.push(entry);
-			}
-			if (display) {
-				this.addHTML(this.styleLeaderboard(leaderboard), true);
-				this.leaderboard.last = leaderboard;
-				this.changed = false;
-				this.lines = { them: 0, total: 0 };
+		const attempts = 3;
+		for (let attempt = 1; attempt <= attempts; attempt++) {
```

Comment:
Why are you making multiple attempts n the consumer as well? We should never need to make 15 attempts here.